testing/bucket_service.go

```diff
@@ -913,6 +961,244 @@ func FindBuckets(
+					},
+				},
+			},
+			args: args{
+				organizationID: idOne,
+				findOptions: influxdb.FindOptions{
+					Limit: 4,
+				},
+			},
+			wants: wants{
+				buckets: []*influxdb.Bucket{
+					{
+						Name:            influxdb.TasksSystemBucketName,
+						ID:              influxdb.TasksSystemBucketID,
+						Type:            influxdb.BucketTypeSystem,
+						RetentionPeriod: influxdb.TasksSystemBucketRetention,
+						Description:     "System bucket for task logs",
+					},
+					{
+						Name:            influxdb.MonitoringSystemBucketName,
+						ID:              influxdb.MonitoringSystemBucketID,
+						Type:            influxdb.BucketTypeSystem,
+						RetentionPeriod: influxdb.MonitoringSystemBucketRetention,
+						Description:     "System bucket for monitoring logs",
+					},
+					{
+						OrgID: idOne,
+						ID:    idTwo,
+						Name:  "aaa",
+					},
+					{
+						OrgID: idOne,
+						ID:    idThree,
+						Name:  "abc",
+					},
+				},
+			},
+		},
+		{
+			name: "do not create duplicate legacy task and monitoring buckets",
+			fields: BucketFields{
+				OrgIDs:    mock.NewIncrementingIDGenerator(idOne),
+				BucketIDs: mock.NewIncrementingIDGenerator(idOne),
+				Organizations: []*influxdb.Organization{
+					{
+						// ID(1)
+						Name: "theorg",
+					},
+				},
+				Buckets: []*influxdb.Bucket{
+					{
+						Name:            influxdb.MonitoringSystemBucketName,
+						OrgID:           idOne,
+						ID:              influxdb.MonitoringSystemBucketID,
+						Type:            influxdb.BucketTypeSystem,
+						RetentionPeriod: influxdb.MonitoringSystemBucketRetention,
+						Description:     "System bucket for monitoring logs",
+					},
+					{
+						Name:            influxdb.TasksSystemBucketName,
+						OrgID:           idOne,
+						ID:              influxdb.TasksSystemBucketID,
+						Type:            influxdb.BucketTypeSystem,
+						RetentionPeriod: influxdb.TasksSystemBucketRetention,
+						Description:     "System bucket for task logs",
+					},
+					{
+						OrgID: idOne,
+						ID:    idTwo,
+						Name:  "abc",
+					},
+				},
+			},
+			args: args{
+				organizationID: idOne,
+			},
+			wants: wants{
+				buckets: []*influxdb.Bucket{
+					{
+						Name:            influxdb.MonitoringSystemBucketName,
+						ID:              influxdb.MonitoringSystemBucketID,
+						OrgID:           idOne,
+						Type:            influxdb.BucketTypeSystem,
+						RetentionPeriod: influxdb.MonitoringSystemBucketRetention,
+						Description:     "System bucket for monitoring logs",
+					},
+					{
+						Name:            influxdb.TasksSystemBucketName,
+						ID:              influxdb.TasksSystemBucketID,
+						OrgID:           idOne,
+						Type:            influxdb.BucketTypeSystem,
+						RetentionPeriod: influxdb.TasksSystemBucketRetention,
+						Description:     "System bucket for task logs",
+					},
+					{
+						ID:    idTwo,
+						OrgID: idOne,
+						Name:  "abc",
+					},
+				},
+			},
+		},
```

Comment:
This is looking great Alirie.

Just wanted to sync up on this a little later. We've had the split now. So we need to port this work.
Also wanted to talk through adding a case or two to this block. To cover things like pages which shouldn't include the system buckets, but for an org which has system buckets.
Also wanted to talk through some of how things IDs now in tests. Since we use the id generators now, instead of the IDs on the fixtures in the test fields.
This is looking great though.